addons/gddialog/ui/ui_database_editor/ui_hbc_filters/ui-hbc-filters.gd

```diff
@@ -0,0 +1,107 @@
+@tool
+extends HBoxContainer
+
+@onready var nSignals: Node = get_node( "Signals" )
+@onready var nLineEditSearch: LineEdit = get_node( "LineEditSearch" )
+@onready var nOptionButtonSearchBy: OptionButton = get_node( 
+		"OptionButtonSearchBy" )
+@onready var nCheckBoxTagFilter: CheckBox = get_node( "CheckBoxTagFilter" )
+@onready var nMenuButtonFilterTags: MenuButton = get_node( 
+		"MenuButtonFilterTags" )
+
+
+""" Search Features """
+
+
+func set_all_records_visible( value: bool ) -> void:
+	var records = owner.nVBCDialogRecords.get_children()
+	for record in records:
+		record.visible = value
+
+
+func update_filter() -> void:
+	var filter_text: String = nLineEditSearch.text
+	set_all_records_visible( true )
+	#	Fill an array with the tags we have selected to filter by.
+	var selected_tags: Array
+	var nFilterMenu: PopupMenu = nMenuButtonFilterTags.get_popup()
+	for tag in range( nFilterMenu.get_item_count() ):
+		if( nFilterMenu.is_item_checked( tag ) ):
+			var tag_name: String = nFilterMenu.get_item_text( tag )
+			selected_tags.push_back( tag_name )
+	#	Now we gather the records we will be filtering.
+	var records: Array = owner.nVBCDialogRecords.get_children()
+	#	Are we filtering by Name/Description/ID/etc?
+	var filter_option: int = nOptionButtonSearchBy.selected
+	#	Now to do the record filtering.
+	for record in records:
```

Comment:
This feels heavily nested.

Reply:
There's not a style guideline for nesting but I do want to do what I can to reduce that.